submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/migrations/DatabaseChangeLog.kt

```diff
@@ -185,3 +200,40 @@ suspend fun ReactiveMongoOperations.ensureStatsIndexes() {
         ensureIndex(backgroundIndex().on(SUB_ACC_NO, ASC)).awaitSingleOrNull()
     }
 }
+
+/**
+ * Published flag for all the submission files
+ */
+suspend fun ReactiveMongoOperations.publishedFlag() {
+    suspend fun setReleasedFlag(submission: Submission) {
+        val query = Query(
+            where(FILE_LIST_DOC_FILE_SUBMISSION_ACC_NO).`is`(submission.accNo)
+                .andOperator(where(FILE_LIST_DOC_FILE_SUBMISSION_VERSION).`is`(submission.version))
+        )
+
+        updateMulti(
+            query,
+            Update().set("$FILE_LIST_DOC_FILE_FILE.$FIRE_FILE_DOC_PUBLISHED", submission.released),
+            FileListDocFile::class.java
+        ).awaitSingleOrNull()
+    }
+
+    val aggregation = newAggregation(
+        DocSubmission::class.java,
+        match(where(SUB_VERSION).gt(0).andOperator(where(STORAGE_MODE).`is`(StorageMode.FIRE.value))),
+        lookup(FILE_LIST_DOC_FILE_COLLECTION_NAME, SUB_ACC_NO, FILE_LIST_DOC_FILE_SUBMISSION_ACC_NO, "result"),
+        match(where("result").not().size(0)),
+        project(SUB_ACC_NO, SUB_VERSION, SUB_RELEASED)
+    )
+
+    aggregate(aggregation, Submission::class.java)
+        .asFlow()
+        .map { setReleasedFlag(it) }
+        .collect()
```

Comment:
are we not planing to set flag for internal files but only to file list files?

Reply:
Good catch